extensions/wikia/RTE/RTEReverseParser.class.php

```diff
@@ -414,19 +414,21 @@ private function handlePlaceholder($node, $textContent) {
 		$out = $data['wikitext'];
 
 		// extra fixes for different types of placeholders
-		if (isset($data['type'])) {
-			switch($data['type']) {
+		if ( isset( $data['type'] ) ) {
+			switch ( $data['type'] ) {
 				// templates and extension tags which are wrapped in <div> needs to have additional newline at the end
 				case 'double-brackets':
 				case 'ext':
-					if ($node->nodeName === 'div') {
+					if ( $node->nodeName === 'div' &&
+						!self::isFirstChild( $node ) &&
+						!self::previousSiblingIs( $node, [ 'h1', 'h2', 'h3', 'h4', 'h5', 'h6' ] ) ) {
```

Comment:
What about <header> - do we allow this? In theory, this can contain h1-h6 + something inside.

Reply:
I think wikitext markup does not insert <header> but we can add it here if we spot it breaks something